rfcs/0001-inline-expression-value-logging.md

```diff
@@ -0,0 +1,191 @@
+```
+
+For symmetry with the existing `log!` macro, a `logv!` is also
+included, which allows passing a `Level` as parameter.
+
+Finally, like all the other public logging macros, the _-v_ macros
+allows overriding the default module-path target with a string
+literal:
+
+```rust
+let i = 33;
+let j = warnv!(target: "maths", (i-1) / 2);
+assert_eq!(j, 16);
+```
+
+See also the [implementation PR], which includes guide level
+documentation, in the form of rustdoc with doc-tests.
+
+# Alternatives
+
+## Multiple expression support
+
+[RFC 2173] included multiple-expression support for `std::dbg!` but
+was closed, at least in part due to perceived complexity with multiple
+expressions and tuple value returns, in preference to [RFC 2361] which
+was ultimately merged and implemented. While `log` is arguably a "more
+advanced" tool, the complexity of multiple expression support does not
+seem to be warranted by the functional gain or convenience win, over
+using the existing formatted logging macros for more complex
+formatting requirements.
+
+## DSL extension of existing log macros
+
+Instead of adding a new set of _-v_ macros, it would be possible to
+extend the existing logging macros by using some additional marker
+syntax, such as the following:
+
+```rust
+debug!(= n/2)
+trace!("{} = {:x}", =i);
+```
+
+Here the `=` signals that the expression should be `stringify!`-ed for
+the message and its value returned from the macro. This complicates
+the macro's, but more importantly, considerably complicates the
+necessary guide documentation for new and existing users to understand
+an evolving logging _DSL_ as new syntax.  This syntax isn't any more
+compact. It aids comprehension when the macro arguments are as
+function-like as possible, by introducing the _-v_ macros specific to
+the feature as designed.
+
+## Only customize the value part of the format
+
+As proposed above, the entire format string may be customized. Until
+understanding that the single `i` argument is expanded to two
+arguments for formatting, it is surprising to see the following, as
+used above, with two placeholders in the format string:
+
+``` rust
+tracev!("{} = {:x}", i);
+```
+
+A workable alternative would be to only allow customizing the value
+part of the format, and have the expression part be fixed as `"{} =
+"`:
+
+``` rust
+tracev!("{:x}". i);
+```
+
+While this alternative is more compact for the subset of compatible
+use cases, it offers _less_ customization options.  For example,
+supplying a custom literal prefix to the message, for additional
+context. It is also less symmetric with the the _formatted logging_
+macros, where the format string literal represents the _entire_ text
+message of the log `Record`. Finally, this alternative adds a small
+but measurable amount of formatting overhead, as the value needs to be
+formatted to a temporary `fmt::Arguments` before formatting to the
+final message.
+
+## Release this as a separate crate, not in `log`
```

Comment:
This is the big question for me. Are users of the `log` crate going to get enough value out of `logv` to warrant building and supporting a parallel API?

Your point about `debug`, `info`, `warn` etc already being effectively parallel APIs is reasonable though.